pentaho-osgi-utils-impl/src/main/java/org/pentaho/osgi/impl/KarafBlueprintWatcherImpl.java

```diff
@@ -101,7 +106,52 @@ public KarafBlueprintWatcherImpl( BundleContext bundleContext ) {
 
       } catch ( Exception e ) {
         throw new BlueprintWatcherException( "Unknown error in KarafBlueprintWatcher", e );
+      } finally {
+        if ( unloadedAndFailedBlueprints.size() > 0 ) {
+          logger.debug( System.lineSeparator() + getBlueprintsReport( blueprintStateService,
+              unloadedAndFailedBlueprints ) );
+        }
       }
     }
   }
+
+  private String getBlueprintsReport( BlueprintStateService blueprintStateService, List<Bundle> bundles ) {
+    String bundlesReport = "--------- Karaf Blueprint Watcher Report Begin ---------";
+    bundlesReport += System.lineSeparator() + "Blueprint Bundle(s) not loaded:";
+    for ( Bundle bundle : bundles ) {
+      bundlesReport +=
+          System.lineSeparator() + "\t" + getBlueprintReport( blueprintStateService, bundle ).replaceAll( "\n",
+              "\n \t" );
```

Comment:
I didn't use System.lineSeparator() because there are some parts of these Strings that I don't control. For instance, ExceptionUtils.getStackTrace( failureCause ), is it building new lines with System.lineSeparator()? I don't know. 
So, instead of somehow guarantying that each String that I don't control will have the System.lineSeparator() for expressing new lines, I thought it was simpler to just use "\n" for replacing. If you think that the best option is to guarantee that the System.lineSeparator() is used, then I will not have objections, just let me know that and I can do the proper changes. Thanks!